h2/src/test/org/h2/test/db/TestGeneralCommonTableQueries.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2004-2014 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (http://h2database.com/html/license.html).
+ * Initial Developer: H2 Group
+ */
+package org.h2.test.db;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import org.h2.test.TestBase;
+
+/**
+ * Test non-recursive queries using WITH, but more than one common table defined.
+ */
+public class TestGeneralCommonTableQueries extends TestBase {
+
+    private static final String SIMPLE_TWO_COMMON_QUERY = "with " +
+            "t1(n) as (select 1 as first) " +
+            ",t2(n) as (select 2 as first) " +
+            "select * from t1 union all select * from t2";
+    
+    private static final String PARAMETERIZED_TWO_COMMON_QUERY = "with " +
+            "t1(n) as (select 2 as first) " +
+            ",t2(n) as (select 3 as first) " +
+            "select * from t1 union all select * from t2 where n<>?";
+    
+    private static final String PARAMETERIZED_THREE_COMMON_QUERY = "with " +
+            "t1(n) as (select 2 as first) " +
+            ",t2(n) as (select 3 as first) " +
+            ",t3(n) as (select 4 as first) " +
+            "select * from t1 union all select * from t2 union all select * from t3 where n<>?";
+
+    private static final String PARAMETERIZED_THREE_COMMON_QUERY_IMPLIED_COLUMN_NAMES = "with " +
+            "t1 as (select 2 as first_col) " +
+            ",t2 as (select first_col+1 from t1) " +
+            ",t3 as (select 4 as first_col) " +
+            "select * from t1 union all select * from t2 union all select * from t3 where first_col<>?";
+    
+    private static final String PARAMETERIZED_CHAINED_QUERY = "    WITH t1 AS ("
```

Comment:
there is no reason to have these as constants, since they are only referred to once. Also easier to read the tests if they are simply inline